embedded/src/main/java/org/wildfly/core/embedded/EmbeddedProcessFactory.java

```diff
@@ -260,15 +259,14 @@ private static ModuleLoader setupModuleLoader(String modulePath, String... syste
 
         assert modulePath != null : "modulePath not null";
 
-        // verify the supplied modules path exists, and if it does not, stop and allow the user to correct.
+        // verify the the first element of the supplied modules path exists, and if it does not, stop and allow the user to correct.
         // Once modules are initialized and loaded we can't change Module.BOOT_MODULE_LOADER (yet).
 
-        File moduleDir = new File(modulePath);
+        File moduleDir = new File(trimPathToModulesDir(modulePath));
         if (!moduleDir.exists() || !moduleDir.isDirectory()) {
             throw new RuntimeException("The specified module directory " + modulePath + " is invalid or does not exist.");
         }
 
-        WildFlySecurityManager.setPropertyPrivileged(SYSPROP_KEY_JBOSS_MODULES_DIR, trimPathToModulesDir(modulePath));
```

Comment:
As far as I was able to follow, that system property is indeed set and read at some points of the code, but apart from checking that the referenced directory exists, nothing is affected by it.

It is referenced in org.jboss.as.server.ServerEnvironment, but all it does is check if it is set to a real directory, and sets it to a default if it is not set, then stores it back to the property.

It is also referenced in a bunch of other locations, that read, write, verify, and massage the configuration model, but has no actual effect on classloading (or anything).

The PR passes the Embedded test cases and my use case. 
AFAIK wildfly Arquillian smoke tests also use this internally to start/sop the WF copies.

That said, it is still possible that I have missed something, and keeping it it doesn't affect the fix for the original bug (only the readability of the code)